test/powershell/Modules/Microsoft.PowerShell.Core/History.Tests.ps1

```diff
@@ -108,4 +108,44 @@
 
         $errorResult | Should be CommandNotFoundException
     }
+
+    It "Has time view in table format" {
+        $null = Get-ChildItem
+        Get-History | Format-Table -View time -ErrorVariable ftError | Out-Null
+        $ftError.Count | Should Be 0
+    }
+
+    It "HistoryInfo calculates Duration" {
+        $ctor = [Microsoft.PowerShell.Commands.HistoryInfo].GetConstructors([Reflection.BindingFlags]::NonPublic -bor [Reflection.BindingFlags]::Instance).Where{$_.GetParameters().Length -eq 5}
+        $start = [datetime]::new(2001, 01, 01, 10, 01, 01)
+        $duration = [timespan] "1:2:21"
+        $end = $start + $duration
+        $hi = $ctor.Invoke((1, "command", [Management.Automation.Runspaces.PipelineState]::Completed, $start, $end))
+        $hi.Duration | Should Be $duration
+    }
+
+    It "Displays HistoryInfo with Duration <expectedResult> for timespan <duration>" -TestCases @(
+        @{duration = "0:0:21"; expectedResult = "00:21.000"}
+        @{duration = "1:14:21"; expectedResult = "01:14:21"}
+        @{duration = "1.1:14:21"; expectedResult = "1.01:14:21"}
```

Comment:
`TimeSeparator` can be culture sensitive. Could you please use TimeSpan constructor for `duration`?

Also in C# `$@"{duration:d\.hh\:mm\:ss}";` should we be culture sensitive too?

Reply:
@iSazonov Excellent catch!
I will fix!

Reply:
Closed.